scripts/create-custom-app.sh

```diff
@@ -0,0 +1,202 @@
+
+# Parse arguments
+if [ $# -lt 3 ]; then
+  echo "Usage: $0 <app-name> <docker-image> <port> [username] [home-dir]"
+  echo ""
+  echo "Arguments:"
+  echo "  app-name      - Name of your custom app (e.g., my-jupyter-app)"
+  echo "  docker-image  - Docker image to use (e.g., jupyter/base-notebook)"
+  echo "  port          - Port your app exposes (e.g., 8888)"
+  echo "  username      - (Optional) User inside container (default: root)"
+  echo "  home-dir      - (Optional) Home directory (default: /root or /home/<username>)"
+  echo ""
+  echo "Example:"
+  echo "  $0 my-jupyter jupyter/base-notebook 8888 jovyan /home/jovyan"
+  exit 1
+fi
+
+readonly APP_NAME="$1"
+readonly DOCKER_IMAGE="$2"
+readonly PORT="$3"
+readonly USERNAME="${4:-root}"
+
+# Calculate home directory if not provided
+if [ $# -ge 5 ]; then
+  readonly HOME_DIR="$5"
+else
+  if [ "$USERNAME" = "root" ]; then
+    readonly HOME_DIR="/root"
+  else
+    readonly HOME_DIR="/home/$USERNAME"
+  fi
+fi
+readonly -f
+
+readonly APP_DIR="src/${APP_NAME}"
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+readonly SCRIPT_DIR
+REPO_ROOT="$(cd "${SCRIPT_DIR}/.." && pwd)"
+readonly REPO_ROOT
+
+# Create app directory
+echo "Creating app directory: ${APP_DIR}"
+mkdir -p "${REPO_ROOT}/${APP_DIR}"
+
+# Generate .devcontainer.json
+echo "Generating .devcontainer.json"
+cat > "${REPO_ROOT}/${APP_DIR}/.devcontainer.json" <<EOF
+{
+  "name": "${APP_NAME}",
+  "dockerComposeFile": "docker-compose.yaml",
+  "service": "app",
+  "shutdownAction": "none",
+  "workspaceFolder": "/workspace",
+  "postCreateCommand": [
+    "./startupscript/post-startup.sh",
+    "${USERNAME}",
+    "${HOME_DIR}",
+    "\${templateOption:cloud}"
+  ],
+  "postStartCommand": [
+    "./startupscript/remount-on-restart.sh",
+    "${USERNAME}",
+    "${HOME_DIR}",
+    "\${templateOption:cloud}"
+  ],
+  "features": {
+    "ghcr.io/devcontainers/features/java:1": {
+      "version": "17"
+    },
+    "ghcr.io/devcontainers/features/aws-cli:1": {},
+    "ghcr.io/dhoeric/features/google-cloud-cli:1": {}
+  },
+  "remoteUser": "root"
+}
+EOF
+
+# Generate docker-compose.yaml
+echo "Generating docker-compose.yaml"
+cat > "${REPO_ROOT}/${APP_DIR}/docker-compose.yaml" <<EOF
+services:
+  app:
+    # The container name must be "application-server"
+    container_name: "application-server"
+    # This can be either a pre-existing image or built from a Dockerfile
+    image: "${DOCKER_IMAGE}"
+    # build:
+    #   context: .
+    restart: always
+    volumes:
+      - .:/workspace:cached
+      - work:${HOME_DIR}/work
+    # The port specified here will be forwarded and accessible from the
+    # Workbench UI.
+    ports:
+      - ${PORT}:${PORT}
+    # Start ttyd with writable mode (-W flag) for interactive terminal
+    # Container runs as root for capabilities, but terminal runs as configured user
+    command: ["ttyd", "-W", "-p", "${PORT}", "su", "-", "${USERNAME}"]
```

Comment:
Should we include the ttyd feature if we're defaulting to start ttyd?

Reply:
i added a ubuntu-example/ for the ttyd usecase.